diff --git a/components/wd-message-box/types.ts b/components/wd-message-box/types.ts
index 5ce339c8a545a615a4ecfff94d0ca6a43d10c56a..81f038ad6da0902f12dc871fac2a4467969f885c 100644
--- a/components/wd-message-box/types.ts
+++ b/components/wd-message-box/types.ts
@@ -14,7 +14,7 @@ import { type InputSize, type InputType } from '../wd-input/types'
 export type MessageType = 'alert' | 'confirm' | 'prompt'
 
 export type MessageBeforeConfirmOption = {
-  resolve: (isPass: boolean) => void
+  resolve: (isPass?: boolean) => void
 }
 
 export type MessageOptions = {
diff --git a/components/wd-message-box/wd-message-box.vue b/components/wd-message-box/wd-message-box.vue
index 354b8fbb926af6dbf5427541e88e3e91f1e5c843..a96643d66760555d7c6ca660a0dfc9f836c2cf1d 100644
--- a/components/wd-message-box/wd-message-box.vue
+++ b/components/wd-message-box/wd-message-box.vue
@@ -166,7 +166,18 @@ function toggleModal(action: 'confirm' | 'cancel' | 'modal') {
       if (messageState.beforeConfirm) {
         messageState.beforeConfirm({
           resolve: (isPass) => {
-            if (isPass) {
+            const localConfirmButtonText = messageState.confirmButtonText
+            const localShowCancelButton = messageState.showCancelButton
+            const localCloseOnClickModal = messageState.closeOnClickModal
+            messageState.confirmButtonProps = { ...(messageState.confirmButtonProps ?? {}), loading: true }
+            messageState.confirmButtonText = "加载中..."
+            messageState.showCancelButton = false
+            messageState.closeOnClickModal = false
+            if (isDef(isPass) || isPass) {
+              messageState.confirmButtonProps.loading = false
+              messageState.confirmButtonText = localConfirmButtonText
+              messageState.showCancelButton = localShowCancelButton
+              messageState.closeOnClickModal = localCloseOnClickModal
               handleConfirm({
                 action: action,
                 value: messageState.inputValue
